### PR TITLE
post: what's new in angular 2.3 and 2.4

### PR DESCRIPTION
**Pull request by cexbrayat**

### CODE REVIEW

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+---
+layout: post
+title: What's new in Angular 2.3 and 2.4?
+author: cexbrayat
+tags: ["Angular 2", "Angular"]
+description: "Angular 2.3 and 2.4 are out. What new features are included?"
```
Comment (clacote) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: Toujours le doute "what -> which" (moi je dirais "which")

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+---
+
+New month, new minor release of Angular!
+December is the month of the last 2.x releases,
+because the next one will be... Angular 4!
+If you missed the official announcement (Igor Minar's keynote),
```
Comment (clacote) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: Didn't you forget to add a link to "Igor Minar's keynote"? It's strange to name it without link.

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+let me sum it up for you.
+
+We'll have a major release every six months, according to the [plan](http://angularjs.blogspot.fr/2016/10/versioning-and-releasing-angular.html).
+The next major release is plan for March 2017.
```
Comment (clacote) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: plan -> planned

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+It should have been Angular 3, but the Angular router is already with a version number in 3.x (because it has been rewritten several times during Angular development). So to avoid trouble, everything will be bumped to 4.x!
+Angular 3 will never exist, Angular 4 is the next one, with Angular 5 just around the corner.
+And know it should be called just "Angular".
```
Comment (clacote) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: know -> now

Comment (clacote) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: I would say : "and now **the framework** should be called just "Angular""

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+Don't worry, these releases are not as different as Angular 2 was different from AngularJS 1.x. They will _maybe_ contain deprecations and new APIs.
```
Comment (clacote) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: This sentence is too complex. I would say "these releases won't introduce as many breaking changes as Angular 2 did to AngularJS 1.x".

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+Technically Angular 4 is a new major release because it contains a breaking change: TypeScript 2.1+ will be required if you use it, whereas you currently can to use 1.8+. Nothing too scary.
```
Comment (clacote) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: can to use -> can use

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+Back to our 2.3 and 2.4 releases: what's new in these small releases?
+
+## Language service
+
+One of the most exciting feature is not really in Angular itself,
+but this release contains a new module that will be reaaaaally handy : a language service module. This is really similar to what TypeScript offers.
```
Comment (clacote) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: `reaaaaally handy :`  -> ` reaaaaally handy:` (no space before `:`)

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+Here is how it works in VS Code:
+
+<img src="/assets/images/2016-12-22/language-service.gif" />
+
+## Inheritance
+
+You can now use inheritance in your apps.
+
+You already could but the decorators from the parent were ignored:
+now, the "last" decorator (when you list them in the ancestor first order) of each kind will be applied.
```
Comment (clacote) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: ancestor first -> ancestor-first, no?

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    export class ChildPony extends ParentPony {}
+
+## Route reuse strategy
+
+The Angular router try to optimize a few things for you,
+especially when you navigate from a route to itself:
+when you go from `races/12` to `races/13`,
+the router will keep the `RaceComponent` (instead of destroying it and recreating it). This is powerful, but you then need to subscribe to an observable from the router to know when the parameters change, to display the correct race for example.
+
+This is still the default behavior, but you can now turn it off, and ask the router to destroy and recreate your component every time, by implement [a `RouteReuseStrategy`](https://angular.io/docs/ts/latest/api/router/index/RouteReuseStrategy-class.html).
```
Comment (clacote) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: by implement ->  by implementing

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md (hunk 8 above)
Comment (cexbrayat) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: I don't know, it was like this in the commit message...

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+      }
+    }
+
+    // the parent `ngOnInit` will be called
+    @Component({ selector: 'ns-pony'})
+    export class ChildPony extends ParentPony {}
+
+## Route reuse strategy
+
+The Angular router try to optimize a few things for you,
```
Comment (jnizet) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: try --> tries

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+especially when you navigate from a route to itself:
+when you go from `races/12` to `races/13`,
+the router will keep the `RaceComponent` (instead of destroying it and recreating it). This is powerful, but you then need to subscribe to an observable from the router to know when the parameters change, to display the correct race for example.
```
Comment (jnizet) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: keep -> reuse

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+
+If you define the same decorator on the child,
+this decorator will be used
+(there is no fancy property merging from the parent and the child).
+
+If a class inherits from a parent class and does not declare
+a constructor, it inherits the parent class constructor,
+meaning that the dependency injection will be properly done in the parent class.
+
+The lifecycle hooks will also be properly called in the parent class,
```
Comment (jnizet) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: I'd say "The lifecycle hooks defined in the parent class will also be called properly, unless they are overridden in the child class".

File: _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+let me sum it up for you.
+
+We'll have a major release every six months, according to the [plan](http://angularjs.blogspot.fr/2016/10/versioning-and-releasing-angular.html).
+The next major release is planned for March 2017.
+It should have been Angular 3, but the Angular router is already with a version number in 3.x (because it has been rewritten several times during Angular development). So to avoid trouble, everything will be bumped to 4.x!
+Angular 3 will never exist, Angular 4 is the next one, with Angular 5 just around the corner.
+And now the framework should be called just "Angular".
+
+Don't worry, these releases are not a complete rewrite with no backward compatibility like Angular 2 was. They will _maybe_ contain deprecations and new APIs.
+Technically Angular 4 is a new major release because it contains a breaking change: TypeScript 2.1+ will be required if you use it, whereas you currently can use 1.8+. Nothing too scary.
```
Comment (jnizet) [inline _posts/2016-12-22-what-is-new-angular-2.3-and-2.4.md]: I'd say "the minimum version of TypeScript, if you use it, will be 2.1, whereas the current minimum version is 1.8".

**Pull request merged**